nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,52 @@
+class Email {
+    constructor(subject, body) {
+        this.subject = subject;
+        this.body = body;
+    }
+}
+
+class SpamDetector {
+    filterBasedOnUpperCaseCharacters(email) {
+        const bodyOfMail = email.body;
+        const splitBodyOfMail = bodyOfMail.split(' ')
+        const result = splitBodyOfMail.filter(item => item == item.toUpperCase())
+        const percentage = (result.length * 100) / splitBodyOfMail.length
+        if (percentage > 60) {
+            return true
+        }
+        return false
+
+    }
+    filterBasedOnWords(email) {
+        const spamWords = ['viagra', 'offer', 'free']
+        const bodyOfMail = email.body;
+        const splitBodyOfMail = bodyOfMail.split(' ')
+        const result = splitBodyOfMail.some(item => spamWords.includes(item.toLowerCase()))
+        return result;
+    }
+    filterBasedOnWordHello(email) {
+        const subjectOfMail = email.subject.toLowerCase();
+        if (subjectOfMail == 'hello') {
```

Comment:
you can just say `return subjectOfMail==='hello'`. It will return true if the condition is true.